### PR TITLE
Add Web Components topic

### PR DESCRIPTION
**Pull request by abraham**
- [x] I followed the contributing guidelines: https://github.com/github/explore/blob/master/CONTRIBUTING.md

I am:
  - [ ] Suggesting edits to an existing Topic page
  - [x] Curating a new Topic page

***********EDITING AN EXISTING TOPIC PAGE************

I'm suggesting these edits to an existing topic:
- [ ] Image (and my file is `*.png`, square, dimensions 288x288)
- [ ] Content (and my changes are in `index.md`)

Please explain why these changes are necessary:

************CURATING A NEW TOPIC PAGE************

- [x] I've formatted my changes as a new folder directory, named for the topic as it appears in the URL on GitHub (e.g. `https://github.com/topics/[NAME]`)
- [x] My folder contains a `*.png` image (if applicable) and `index.md`
- [x] All required fields in my `index.md` conform to the Style Guide and API docs: https://github.com/github/explore/tree/master/docs

Please explain why you think this Topic page should be curated:

Web Components are gaining a lot of steam in the community with projects like [Stencil](https://stenciljs.com/), [Vue](https://twitter.com/youyuxi/status/958461422786301957), [Angular Elements](https://blog.angular.io/the-angular-team-at-angularmix-2d56fd7fde65) going all in or showing strong support.

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

### CODE REVIEW

**Comment by abraham** [comment]
Added logo since it doesn't have usage restrictions. https://github.com/webcomponents/webcomponents.org/issues/870

File: topics/web-components/index.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+aliases: webcomponent, web-component, webcomponents
+created_by: Alex Russell
+display_name: Web Components
+github_url: https://github.com/webcomponents
+logo: web-components.png
+related: custom-elements, shadow-dom, es-modules, html-templates, html-imports
+short_description: Web Components are a set of web platform APIs developers can use to create custom HTML tags.
+topic: web-components
+url: https://www.webcomponents.org/
+wikipedia_url: https://en.wikipedia.org/wiki/Web_Components
```
Comment (cheshire137) [inline topics/web-components/index.md]: Based on the Wikipedia page, you could add a `released: 2011` line if you wanted.

File: topics/web-components/index.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+---
+Developers can create custom, reusable, encapsulated HTML tags for the web. Components use the Custom Element standard and frequently are combined with the Shadow DOM, HTML Templates, and ES Modules.
```
Comment (cheshire137) [inline topics/web-components/index.md]: Maybe start the first sentence with "Web Components allow developers to create", or end with "for the web using Web Components."

Would https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements be a good play to link "Custom Element standard" to?

**Comment by abraham** [comment]
Updated based on feedback and included some additional MDN links.

**Comment by abdonrd** [comment]
I'm with @kenchris. Maybe it would be better to remove the `created_by` field.

**Pull request merged**